### PR TITLE
WebGLBindingsStates: Avoid redundant updates when using InstancedMesh.

### PR DESCRIPTION
**Pull request by snagy**
Related issue: None

**Description**

This removes the check that forces a buffer update each frame for instanced meshes.  I'm not sure why this check is necessary (if it's necessary in some situations, I'd love to find a better fix for that!)  After removing this, I validated that all the mesh instancing examples still work, and another example scene I'm building which has a lot of instanced static meshes also works well with far better performance than before.

### CODE REVIEW

**Comment by mrdoob** [comment]
@takahirox Do you remember why that check was there?

**Comment by takahirox** [comment]
IIRC someone else (Maybe @Mugen87 ?) has inserted the lines and I couldn't join the discussion about it because I was busy.

**Comment by Mugen87** [comment]
The fix was added here: #19015

**Comment by snagy** [comment]
Thanks for the example, @Mugen87 !

I made the VAO caching/update logic more closely match the setupVertexAttributes logic; this properly adds the position and color buffers into the cache and fixes the example (and the other instancing examples still work), while still keeping the perf gains.

File: src/renderers/WebGLMultiviewRenderTarget.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+		this.depthBuffer = false;
+		this.stencilBuffer = false;
+
+		this.numViews = numViews;
+	}
+
+	copy( source ) {
+
+		WebGLRenderTarget.prototype.copy.call( this, source );
```
Comment (LeviPesin) [inline src/renderers/WebGLMultiviewRenderTarget.js:21]: Why don't just `super.copy( source )`?

File: src/renderers/WebGLRenderer.js [hunk 2]
```diff
@@ -987,22 +996,40 @@ function WebGLRenderer( parameters = {} ) {
 
 		if ( camera.isArrayCamera ) {
 
-			const cameras = camera.cameras;
 
-			for ( let i = 0, l = cameras.length; i < l; i ++ ) {
+			if ( xr.enabled && multiview.isAvailable() ) {
```
Comment (LeviPesin) [inline src/renderers/WebGLRenderer.js:1000]: Here you can just use `const l = xr.enabled && multiview.IsAvailable() ? 1 : cameras.length`.

File: src/renderers/webgl/WebGLMultiview.js [hunk 3]
```diff
@@ -0,0 +1,595 @@
...
+import { Vector2 } from '../../math/Vector2.js';
+
+
+
+var WGLUProgram = (function() {
+
+  "use strict";
+
+  // Attempts to allow the browser to asynchronously compile and link
+  var Program = function(gl) {
```
Comment (LeviPesin) [inline src/renderers/webgl/WebGLMultiview.js:21]: Just use a class here.

**Comment by snagy** [comment]
Does anyone have any feedback on what they'd like to see tested/updated before this can be integrated?  This change is a huge perf boost for instanced meshes in CPU limited scenes.

**Comment by OndrejSpanel** [comment]
I concur. I was just about to explore why is the forced update there, just because seeing lots of unnecessary data updates in my scene - I am glad a PR is already prepared for that.

**Comment by mrdoob** [comment]
@Mugen87 Should I merge this and see if anything breaks?

**Comment by Mugen87** [comment]
I have it on my to-do list this week to review this PR more closely^^. Just trying to solve another issue first.

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**